2 update docker for app

surebert opened the pull request:
updated requirements to use torch that exists as 2.0.0+cu117 does not exist and therefore gradio app does not work

added docker-compose.yml to allow for docker compose build and docker compose up

switched docker to use the app.py version with advanced options

added example cache directory for models and output samples

upgrading to cuda 12.2 for docker

updated readme.md with updated docker instructions

added default docker cache directories to .gitignore

yujincheng08 commented on Dockerfile:
```diff
@@ -11,22 +11,17 @@ RUN apt-get update && \
         libglib2.0-0 \
     && rm -rf /var/lib/apt/lists/*
-
+    
```
Remove extra space here.

yujincheng08 commented on Dockerfile:
```diff
@@ -11,22 +11,17 @@ RUN apt-get update && \
+    
 ADD requirements.txt .
 
-RUN pip install torch==2.0.0+cu117 torchvision==0.15.1+cu117 torchaudio==2.0.1 --index-url https://download.pytorch.org/whl/cu117 && \
```
we may only need to switch to `pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118`. Cuda 12 requires higher version of driver, which may be a concern.

yujincheng08 replied:
BTW we can specify pypi source from requirements.txt, so we can keep `RUN pip install -r requirements.txt` but instead update the requirements.txt with proper pypi source.

yujincheng08 commented on docker-compose.yml:
```diff
@@ -0,0 +1,24 @@
+version: "3.8"
```
I think it's better to write it in README.

surebert commented:
@yujincheng08 Good catch on devel. I made the changes to runtime version of nvidia base model.  I was compilng stuff in side there in my testing which is why I left devel in before by accident.

I also removed whitespace and updated instructions.  

I left version 12.  It runs really fast on my 4090 and it needs a 4090 at least on the consumer side since it takes 17GB to run.  We could add instructions on how to switch to 118 and 11.8 or we could add a second Dockerfile?

![image](https://github.com/PixArt-alpha/PixArt-alpha/assets/1123857/5213f62c-450d-422e-a4e0-e3cfa19f4b9d)

Out of the box this version runs where the current master branches does not

surebert reviewed (commented):
I think this is ready to go, tested and working

surebert reviewed (commented):
added ability to easily switch between 1024, 512 and LCM version of app in docker with new env varibale APP_CONTEXT

ignoring docker cache directory from build

surebert reviewed (commented):
just updating readme with docker options for switching APP_CONTEXT between 1024, 512 and LCM versions

surebert reviewed (commented):
updated readme

surebert reviewed (commented):
Do you need something else from me to merge this?

lawrence-cj commented:
Nice work. Approved.
